### ISSUE TITLE
Insonsitencies with usage of _WIN32_WINNT

### ISSUE THREAD

**Issue by edo9300** [openssl/openssl#24367]
Openssl has some logic in e_os.h to define the macro `_WIN32_WINNT` if not set by externally by the compiler, and then takes care of including `windows.h`.
https://github.com/openssl/openssl/blob/87314d24c4f025df1ebf47dc527cc8a96bef354a/include/internal/e_os.h#L86-L101
Supposedly this file should be the only one to be used internally by openssl to include windows headers, but in various places windows.h is included directly on top of the file, for example https://github.com/openssl/openssl/blob/87314d24c4f025df1ebf47dc527cc8a96bef354a/crypto/threads_win.c#L10-L15
In other cases, depending on other internal headers included beforehand, either ``windows.h`` is manually included first, or it's included through `e_os.h` due to chain dependencies, in this case `e_os.h` is included by `time-h` https://github.com/openssl/openssl/blob/87314d24c4f025df1ebf47dc527cc8a96bef354a/include/internal/thread_arch.h#L12-L18
These inconsistencies could lead to some files being possibly built with `_WIN32_WINNT` defined as `0x501` (if e_os is inclded first and it's not changed externally), while others being built with whatever the current sdk sets as default.

**Comment by t8m** [comment]
Yeah, `windows.h` should not be included directly. It should always be included via `internal/e_os.h`.

**Comment by t8m** [comment]
Fix was merged.
